gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -322,54 +328,99 @@ def get_summary_stats_filter_groups_ht(
     return ht.filter(ht._no_lcr).drop("_no_lcr")
 
 
-def create_per_sample_counts_ht(
-    mt: hl.MatrixTable, filter_group_ht: hl.Table, autosomes_only: bool = False
-) -> hl.Table:
+def prepare_mt_for_sample_counts(
+    mt: hl.MatrixTable,
+    filter_group_ht: hl.Table,
+    autosomes_only: bool = False,
+    for_intermediate: bool = False,
+) -> hl.MatrixTable:
     """
-    Create Table of Hail's sample_qc output broken down by requested variant groupings.
+    Prepare MatrixTable for computing per-sample stats counts.
 
-    Useful for finding the number of variants per sample, either all variants, or
-    variants fall into specific capture regions, or variants that are rare
-    (adj AF <0.1%), or variants categorized by predicted consequences in all, canonical
-    or mane transcripts.
+    This function prepares the MatrixTable for computing per-sample stats counts. It
+    does the following:
+
+        - Annotates the rows with the filter group metadata from `filter_group_ht`.
+        - If `autosomes_only` is True, filters to autosomes only and performs the high
+          AB het -> hom alt adjustment of GT, or adds a high_ab_het_ref annotation if
+          `for_intermediate` is True.
+        - Filters to `adj` genotypes and selects only the necessary entries for
+          downstream processing.
 
     :param mt: Input MatrixTable containing variant data. Must have multi-allelic sites
         split.
-    :param filter_group_ht: Table containing filter groups for summary stats.
-    :param autosomes_only: Whether to restrict analysis to autosomes only. Default is
-        False.
-    :return: Table containing per-sample variant counts.
+    :param filter_group_ht: Table containing filter group metadata from
+        `get_summary_stats_filter_groups_ht`.
+    :param autosomes_only: Boolean indicating whether to filter to autosomes only. When
+        True, the high AB het -> hom alt adjustment of GT is performed.
+    :param for_intermediate: Boolean indicating whether to prepare the MT for
+        intermediate Table creation (`create_intermediate_mt_for_sample_counts`).
+        Default is False.
+    :return: MatrixTable prepared for computing per-sample stats counts.
     """
-    # Add Allele Type annotations to variant MatrixTable.
-    _, variant_types = vmt_sample_qc_variant_annotations(
-        global_gt=mt.GT, alleles=mt.alleles
-    )
-    mt = mt.annotate_rows(variant_atypes=variant_types)
+    select_fields = ["GT", "GQ", "DP"]
+    if for_intermediate:
```

Comment:
yeah, I will make updates for all the autosome only stuff, for now I think it's OK if it just fails without that flag since that will be the next step that I get in